numpy/core/tests/test_multiarray.py

```diff
@@ -4588,17 +4588,20 @@ def test_out_overlap(self):
 
 class TestLexsort(object):
     def test_basic(self):
-        a = [1, 2, 1, 3, 1, 5]
-        b = [0, 4, 5, 6, 2, 3]
-        idx = np.lexsort((b, a))
-        expected_idx = np.array([0, 4, 2, 1, 3, 5])
-        assert_array_equal(idx, expected_idx)
-
-        x = np.vstack((b, a))
-        idx = np.lexsort(x)
-        assert_array_equal(idx, expected_idx)
-
-        assert_array_equal(x[1][idx], np.sort(x[1]))
+        for dtype in ['uint8', 'uint16', 'uint32', 'uint64',
+                      'int8', 'int16', 'int32', 'int64',
+                      'float32', 'float64']:
+            a = np.array([1, 2, 1, 3, 1, 5], dtype=dtype)
+            b = np.array([0, 4, 5, 6, 2, 3], dtype=dtype)
+            idx = np.lexsort((b, a))
+            expected_idx = np.array([0, 4, 2, 1, 3, 5])
+            assert_array_equal(idx, expected_idx)
+
+            x = np.vstack((b, a))
+            idx = np.lexsort(x)
+            assert_array_equal(idx, expected_idx)
+
+            assert_array_equal(x[1][idx], np.sort(x[1]))
```

Comment:
This becomes `assert_array_equal(a[idx], np.sort(a))`, and the three lines above it can be deleted